waspc/data/Generator/templates/universal/validators.js

```diff
@@ -0,0 +1,21 @@
+export function isValidAbsoluteURL(rawUrl) {
+    try {
+        const url = new URL(rawUrl);
+        /*
+            URL constructor will parse some invalid absolute URLs as valid URLs
+            so we need to do some extra checks.
+
+            Example: "localhost:3000" will be parsed as URL with protocol of
+            "localhost:" and host of "3000"
+        */
+        return url.protocol === "http:" || url.protocol === "https:";
+    } catch (e) {
+        return false;
+    }
+}
+
+export function throwIfNotValidAbsoluteURL(rawUrl, errorMessage) {
+    if (rawUrl && !isValidAbsoluteURL(rawUrl)) {
+        throw new Error(errorMessage);
+    }
```

Comment:
When `rawUrl` is `undefined`, we don't throw an error? Is this expected behavior?

Reply:
This is indeed the expected behaviour since we have fallbacks for `REACT_APP_API_URL` and `WASP_WEB_CLIENT_URL` env variables. If the user doesn't specify the env variables, it should not throw the erorr.

For this use case, this is fine IMHO, if we start using this function more, maybe we can refactor it to be more explicit about not throwing in that case or change the behaviour.